evalml/tests/dependency_update_check/make_deps_diff.sh

```diff
@@ -1,4 +1,4 @@
 reqs_list=$(python -c "import os; from evalml.utils import get_evalml_pip_requirements; print('\n'.join(get_evalml_pip_requirements(os.getcwd())))")
-allow_list=$(echo ${reqs_list} | grep -oE "[a-zA-Z]+[a-zA-Z_\-]*" | paste -d "|" -s -)
+allow_list=$(echo ${reqs_list} | grep -oE "[a-zA-Z]+[a-zA-Z_\-]*" | grep -v "post" | tr "_" "-" | paste -d "|" -s -)
```

Comment:
Added `tr` to replace underscores with hyphens since [pip does that automatically apparently](https://pybuilder.io/documentation/publishing-plugins#:~:text=Pip%20replaces%20underscores%20with%20dashes,a%20syntax%20error%20at%20import.). Might also explain why the deps checker for `category-encoders` wasn't working and why we were a few versions behind.